src/passes/OptimizeInstructions.cpp

```diff
@@ -3604,11 +3604,12 @@ struct OptimizeInstructions
     // Check left's max bits and right is constant.
     auto leftMaxBits = Bits::getMaxBits(left, this);
     uint64_t maskLeft;
-    if (!left->type.isNumber() || leftMaxBits == left->type.getByteSize() * 8) {
+    if (!left->type.isNumber() || leftMaxBits >= left->type.getByteSize() * 8) {
```

Comment:
Perhaps it would be simpler to exit if unreachable? That seems better to me, and would avoid the `>=` here that seems odd without a comment.

Reply:
Since the `!left->type.isNumber()` check isn't sufficient to avoid this, I don't think checking for unreachable here will work. The operand is optimized into a block ending with `unreachable`, but the block itself must have a concrete type.

Reply:
I see, this happens in the middle of the pass, before we refinalize, then.

Then this lgtm, with a comment here explaining that (otherwise the `>=` really looks odd, I think, as I said).